src/clr-core/internal/decorators/property.ts

```diff
@@ -58,16 +65,56 @@ export function getDefaultOptions(
   }
 }
 
+export function requirePropertyCheck(protoOrDescriptor: any, name: string, options?: PropertyConfig) {
+  const targetConnectedCallback: () => void = protoOrDescriptor.connectedCallback;
+
+  function connectedCallback(this: any): void {
+    if (options && options.required) {
+      const message = options.requiredMessage || getRequiredMessage(options.required, name, this.tagName);
+      if (options.required === 'error') {
+        throw new Error(message);
+      } else {
+        LogService.warn(message, this);
+      }
+    }
+
+    if (targetConnectedCallback) {
+      targetConnectedCallback.apply(this);
+    }
+  }
+
+  protoOrDescriptor.connectedCallback = connectedCallback;
+}
+
+function getRequiredMessage(level = 'warning', propertyName: string, tagName: string) {
+  const tag = tagName.toLocaleLowerCase();
+  return (
+    `${capitalizeFirstLetter(
+      level
+    )}: ${propertyName} is required to use ${tag} component. Set the JS Property or HTML Attribute.\n\n` +
+    `${getAngularVersion() ? `Angular: <${tag} [${propertyName}]="..."></${tag}>\n` : ''}` +
+    `${getVueVersion() ? `Vue: <${tag} :${propertyName}="..."></${tag}>\n` : ''}` +
+    `${getReactVersion() ? `React: <${kebabCaseToPascalCase(tag)} ${propertyName}={...} />\n` : ''}` +
+    `${`HTML: <${tag} ${camelCaseToKebabCase(propertyName)}="..."></${tag}>\n`}` +
+    `${`JavaScript: document.querySelector('${tag}').${propertyName} = '...';\n\n`}`
```

Comment:
Here if a property is required and missing we can warn with a code hint. This will adjust based on possible assumed application env.

<img width="903" alt="core-warning" src="https://user-images.githubusercontent.com/2021067/79759769-b8a1fe80-82e4-11ea-9263-f0524007ba9d.png">